pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol

```diff
@@ -921,6 +937,117 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, ReentrancyG
         }
     }
 
+    // Circuit Breakers
+
+    /**
+     * @notice Return the circuit breaker parameters for the given token.
+     * @dev These are the reference values (BPT price and weight complement) computed when the breaker was set,
+     * along with the percentage bounds. To get the current BPT price boundaries needed to check whether the
+     * circuit breaker should trip, call `getCurrentCircuitBreakerBounds`.
+     */
+    function getCircuitBreakerFields(IERC20 token)
+        external
+        view
+        returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return CircuitBreakerLib.getCircuitBreakerFields(_circuitBreakerState[token]);
+    }
+
+    /**
+     * @notice Get the current BPT price bounds for the given token.
+     * @dev The bounds are set as percentages, but retrieved as BPT prices which can be directly compared to the
+     * current BPT price. Though the percentages are fixed when the breaker is set, the BPT price bounds are dynamic.
+     * If there is an ongoing gradual weight update, they will shift up or down with the current weight.
+     */
+    function getCurrentCircuitBreakerBounds(IERC20 token) public view returns (uint256, uint256) {
+        return
+            CircuitBreakerLib.getCurrentCircuitBreakerBounds(
+                _circuitBreakerState[token],
+                _getNormalizedWeight(token).complement()
+            );
+    }
+
+    /**
+     * @notice Returns flags indicating whether the lower or upper bound circuit breakers have tripped.
+     */
+    function hasCircuitBreakerTripped(IERC20 token) external view returns (bool, bool) {
+        return
+            CircuitBreakerLib.hasCircuitBreakerTripped(
+                _circuitBreakerState[token],
+                getActualSupply(),
+                _getNormalizedWeight(token),
+                _getUpscaledTokenBalance(token)
+            );
+    }
```

Comment:
This is a bit of a weird function - I imagine it'd almost always return false, since the whole point of circuit breakers is to prevent them from tripping by rejecting bad swaps. 

The only way this could return true is if the weights change to such a value that the current balances are bad. But even then, that is not useful information: it is possible to construct a swap that goes in the opposite direction and is accepted by the breaker.

Given this I'd just remove this function, as it is misleading and not very useful. Am I missing something?

Reply:
I'd agree with this. The library function is useful for within swaps/joins/exits but probably isn't worth exposing like this.

Reply:
If it's not useful for aggregators, etc., I can remove it. They could do their own calculations based on the settings they observe from the events.

Reply:
Yeah, it's more important to have a "what's the biggest trade I can do" function but that can be in a separate contract.